src/journal/bsod.c

```diff
@@ -188,11 +204,20 @@ static int display_emergency_message_fullscreen(const char *message) {
         if (r < 0)
                 return log_warning_errno(r, "Failed to write emergency message to terminal: %m");
 
+        if (message_id) {
+                char buff[STRLEN("\nScan the QR code below, and search for the entry with the ID: --") + strlen(message_id) + 1];
+                xsprintf(buff, "\nScan the QR code below, and search for the entry with the ID: --%s", message_id);
+
+                r = loop_write(fd, buff, SIZE_MAX, /* do_poll = */false);
+                if (r < 0)
+                        return log_warning_errno(r, "Failed to write emergency message to terminal: %m");
```

Comment:
It is not necessary to combine the two string, but calling loop_write() twice is enough.
```c
r = loop_write(fd, "\nScan the QR ....", SIZE_MAX, /* do_poll = */ false);
if (r < 0)
        return log_warning_errno(r, "Failed to ...");

r = loop_write(fd, message_id, SIZE_MAX, /* do_poll = */ false);
if (r < 0)
        return log_warning_errno(...);
```